### ISSUE TITLE
driver can not handle pseudo types (e.g. 'record') as part of function return value (this makes handling multi-result return values painful)

### ISSUE THREAD

**Issue by pertl** [impossibl/pgjdbc-ng#255]
I call a stored pl/pgsql function via

```
    Connection connection = DriverManager.getConnection("jdbc:pgsql:foo", "foo", "bar");
    connection.createStatement().execute("SELECT get4()");
```

the function basically looks like this

```
CREATE FUNCTION get4() RETURNS RECORD AS
$BODY$ 
DECLARE
  r1     RECORD;
  r2     RECORD;
  result RECORD; 
BEGIN
  SELECT array_agg(c.*) AS arr
  FROM customers AS c
  INTO r1;

  SELECT array_agg(o.*) AS arr
  FROM orders AS o
  INTO r2;

  SELECT
    r1.arr,
    r2.arr
  INTO result;

  RETURN result;

END;
$BODY$
LANGUAGE plpgsql STABLE;
```

executing the test code yields the following exception:

```
java.lang.ClassCastException: com.impossibl.postgres.types.PseudoType cannot be cast to com.impossibl.postgres.types.CompositeType

at com.impossibl.postgres.system.procs.Records$BinDecoder.decode(Records.java:70)
at com.impossibl.postgres.system.procs.BinaryDecoder.decode(BinaryDecoder.java:44)
at com.impossibl.postgres.protocol.v30.BindExecCommandImpl$BindExecCommandListener.rowData(BindExecCommandImpl.java:113)
at com.impossibl.postgres.protocol.v30.ProtocolImpl.receiveRowData(ProtocolImpl.java:1000)
at com.impossibl.postgres.protocol.v30.ProtocolImpl.dispatch(ProtocolImpl.java:790)
at com.impossibl.postgres.protocol.v30.MessageHandler.channelRead(MessageHandler.java:46)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:307)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:293)
at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:276)
at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:354)
at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:244)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:307)
at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:293)
at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:840)
at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:131)
at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:511)
at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:468)
at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:382)
at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:354)
at io.netty.util.concurrent.SingleThreadEventExecutor$2.run(SingleThreadEventExecutor.java:112)
at java.lang.Thread.run(Thread.java:745)
```

in the debuggger the line causing the exception looks like this:

```
@Override
public Object decode(Type type, Short typeLength, Integer typeModifier, ByteBuf buffer, Context context) throws IOException {

  CompositeType compType = (CompositeType) type;   // ==> type: "record(2249)"
```

variable "type" is of type=com.impossibl.postgres.types.PsuedoType which obviously is no composite type.

could you fix that pseudo types work. imho it's far too general to use pseudo types to not support them :-)

this issue exists with 0.6 and with HEAD

postgresql = 9.5.3

thanks!!

cheers
Peter Ertl

**Comment by jesperpedersen** [comment]
Could you provide a patch for this ?

**Comment by pertl** [comment]
I wish I could :-)

unfortunately I don't understand enough about pseudo types and the DataRow result type to do it.

**Comment by pertl** [comment]
as the docs state most languages only use 'void' and 'record' as pseudo return types. so probably just supporting these two types would already be beneficial (https://www.postgresql.org/docs/9.5/static/datatype-pseudo.html#DATATYPE-PSEUDOTYPES-TABLE).

**Comment by kdubb** [comment]
Fixed in #264
